torch/_dynamo/variables/builtin.py

```diff
@@ -432,7 +432,9 @@ def constant_args(self, *args, **kwargs):
 
     def tensor_args(self, *args, **kwargs):
         return any(
-            isinstance(i, variables.TensorVariable)
+            isinstance(
+                i, (variables.TensorVariable, variables.TensorWithTFOverrideVariable)
```

Comment:
you called out the lack of inheritance in your design doc too. Why isn't the right thing to do to fix the inheritance hierarchy here first, and THEN do the rest of the PR

Reply:
So this is bit of a chicken and egg problem - the refactoring I did makes creating the inheritance hierarchy easier, but requires me to also touch a lot of tensor wrapping to get this to work. I think this would be best to do separately.